examples/infill/infill.cpp

```diff
@@ -376,7 +376,7 @@ int main(int argc, char ** argv) {
                     n_past, n_left, n_ctx, params.n_keep, n_discard);
 
                 llama_kv_cache_seq_rm (ctx, 0, params.n_keep + 1            , params.n_keep + n_discard + 1);
-                llama_kv_cache_seq_add(ctx, 0, params.n_keep + 1 + n_discard, n_past, -n_discard);
+                llama_kv_cache_seq_add(ctx, 0, params.n_keep + 1 + n_discard, n_past + 1,    -n_discard);
```

Comment:
Small explanation for what's happening: We suppose to shift all tokens from `n_keep + n_discard + 1`, so the end of must be `n_past + 1` (or we can simply set `-1` here)

Reply:
Hm, I don't think `n_past + 1` is needed here. There shouldn't be a token with `pos == n_past` in the KV cache.

But yes, using either `n_past` or `-1` would achieve the same thing. Thing using `n_past` is more illustrative.